src/IB/IIMethod.cpp

```diff
@@ -846,7 +898,8 @@ IIMethod::interpolateVelocity(const int u_data_idx,
         NumericVector<double>* U_n_vec = nullptr;
         NumericVector<double>* U_t_vec = nullptr;
         NumericVector<double>* X_vec = nullptr;
-        NumericVector<double>* X_ghost_vec = d_X_IB_ghost_vecs[part];
```

Comment:
I think that you will still want to use `d_X_IB_ghost_vecs` here? `X_IB_ghosts` is just a vector with DOFs set up appropriately to manage interaction between the boundary mesh and the background Cartesian grid.